ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/Genesis/Tests/Uniform.hs

```diff
@@ -228,47 +237,39 @@ prop_leashingAttackStalling =
       advs <- mapM dropRandomPoints $ adversarialPeers sch
       pure $ ps {psSchedule = sch {adversarialPeers = advs}}
 
-    disableBoringTimeouts gt =
-      gt { gtChainSyncTimeouts = (gtChainSyncTimeouts gt)
-            { mustReplyTimeout = Nothing
-            , idleTimeout = Nothing
-            }
-         }
-
-    dropRandomPoints :: [(Time, SchedulePoint blk)] -> QC.Gen [(Time, SchedulePoint blk)]
-    dropRandomPoints ps = do
+dropRandomPoints :: [(Time, SchedulePoint blk)] -> QC.Gen [(Time, SchedulePoint blk)]
+dropRandomPoints ps = do
       let lenps = length ps
-      dropCount <- QC.choose (0, max 1 $ div lenps 5)
+          dropsMax = max 1 $ lenps - 1
+      dropCount <- QC.choose (div dropsMax 2, dropsMax)
       let dedup = map NE.head . NE.group
       is <- fmap (dedup . sort) $ replicateM dropCount $ QC.choose (0, lenps - 1)
       pure $ dropElemsAt ps is
-
+  where
     dropElemsAt :: [a] -> [Int] -> [a]
-    dropElemsAt xs [] = xs
-    dropElemsAt xs (i:is) =
-      let (ys, zs) = splitAt i xs
-       in ys ++ dropElemsAt (drop 1 zs) is
+    dropElemsAt xs is' =
+      let is = Set.fromList is'
+      in map fst $ filter (\(_, i) -> not $ i `Set.member` is) (zip xs [0..])
```

Comment:
@nbacquey List comprehensions seem to be more readable, more compact and won't rely on list fusion to be efficient.

Also switching to `IntSet` will be more efficient and there are special functions in containers `notMember x s`, that can be used instead of `not $ member x s`
```suggestion
      let is = IntSet.fromList is'
      in [x | (x, i) <- zip xs [0..], i `IntSet.notMember` is]
```